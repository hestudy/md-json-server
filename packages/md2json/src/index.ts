import remarkFrontmatter from "remark-frontmatter";
import remarkParse from "remark-parse";
import { unified } from "unified";
import yaml from "yaml";
import { removePosition } from "unist-util-remove-position";

export const toJson = async (md: string) => {
  const processor = unified().use(remarkParse).use(remarkFrontmatter);

  const parseTree = processor.parse(md);
  const tree = await processor.run(parseTree);

  removePosition(tree, { force: true });

  return tree;
};
