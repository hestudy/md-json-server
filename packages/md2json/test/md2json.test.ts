import { expect, it } from "vitest";
import { toJson } from "../src";

const md = `---
title: 测试
date: 2024-02-01
---

# 标题

内容`;

it("toJson", async () => {
  const json = await toJson(md);
  expect(json).toMatchSnapshot();
});
