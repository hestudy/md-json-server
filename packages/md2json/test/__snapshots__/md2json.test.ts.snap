// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`toJson 1`] = `
{
  "children": [
    {
      "type": "yaml",
      "value": "title: demo",
    },
    {
      "children": [
        {
          "type": "text",
          "value": "Pluto",
        },
      ],
      "depth": 1,
      "type": "heading",
    },
    {
      "children": [
        {
          "children": [
            {
              "type": "text",
              "value": "Pluto",
            },
          ],
          "type": "strong",
        },
        {
          "type": "text",
          "value": " (minor-pâ€¦",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;
